assets/wire/components/fields/index.jsx

```diff
@@ -0,0 +1,164 @@
+        if (
+            typeof field.component === 'string' &&
+            typeof item[field.field] === 'string'
+        ) {
+            // example: { field: "version", component: "version_type" }
+            switch (field.component) {
+            case 'version_type':
+                return {
+                    key: field.field,
+                    Component: () => (
+                        <VersionType value={item[field.field]} />
+                    ),
+                };
+            }
+
+            return null;
+        }
+
+        if (typeof field.styles === 'object') {
+            // example: { field: "source", styles: {fontWeight: "bold"} }
+            const result = getComponentForField(item, field.field);
+
+            return result
+                ? {
+                    key: field.field,
+                    Component: (props) => (
+                        <span style={field.styles || {}}>
+                            <result.Component {...props} />
+                        </span>
+                    ),
+                }
+                : null;
+        }
+
+        return null;
+    } else if (Array.isArray(field) && field.length > 0) {
+        // example: ["source", "//", "department"]
+        const components = field
+            .map((f) => getComponentForField(item, f))
+            .filter(Boolean);
+
+        // remove orphan separators. For example in ['source', '//', 'department']
+        // if the 'department' is empty, then '//' should not be shown
+        if (components[components.length - 1].key === SEPARATOR_KEY) {
+            components.pop();
+        }
+
+        return {
+            key: components.map(({key}) => key).join('-'),
+            Component: (props) => (
+                <span>
+                    {components.map(({Component}, i) => (
+                        <Component key={i} {...props} />
+                    ))}
+                </span>
+            ),
+        };
+    } else if (typeof field === 'string') {
+        // example: "//"
+        if (ALLOWED_SEPARATORS.includes(field)) {
+            return {
+                key: 'separator', // will be modified afterwards, as it's not unique
+                Component: () => <span> {field} </span>,
+            };
+        }
+
+        let Component = null;
+
+        // example: "source"
+        switch (field) {
+        case 'urgency':
+            Component = UrgencyLabel;
+            break;
+        case 'source':
+            Component = Source;
+            break;
+        case 'charcount':
+            Component = CharCount;
+            break;
+        case 'wordcount':
+            Component = WordCount;
+            break;
+        case 'previous_versions':
+            Component = PreviousVersions;
+            break;
+        case 'embargo':
+            Component = Embargo;
+            break;
+        case 'versioncreated':
+            Component = VersionCreated;
```

Comment:
imo would be good to have some mapping for components so it would be easy to add a new type or override the core implementation with something custom

Reply:
like an object or something?

```ts
{
  versioncreated: VersionCreated,
  ...
}
```

Reply:
exactly